photon-ml/src/main/scala/com/linkedin/photon/ml/algorithm/CoordinateDescent.scala

```diff
@@ -149,32 +167,35 @@ class CoordinateDescent(
           case rddLike: RDDLike => rddLike.unpersistRDD()
           case _ =>
         }
+
         updatedGAMEModel = updatedGAMEModel.updateModel(coordinateId, updatedModel)
 
         // Summarize the current progress
         logger.info(s"Finished training the model in coordinate $coordinateId, " +
             s"time elapsed: ${modelUpdatingTimer.stop().durationSeconds} (s).")
-        optimizationTrackerOption.foreach { optimizationTracker =>
-          logger.debug(s"OptimizationTracker:\n${optimizationTracker.toSummaryString}")
-          optimizationTracker match {
-            case rddLike: RDDLike => rddLike.unpersistRDD()
-            case _ =>
-          }
-        }
         logger.debug(s"Summary of the learned model:\n${updatedModel.toSummaryString}")
 
+        optimizationTrackerOption.foreach {
+          optimizationTracker => logger.debug(s"OptimizationTracker:\n${optimizationTracker.toSummaryString}")
+            optimizationTrackerOption match {
```

Comment:
Yes, good catch. Also `optimizationTracker =>` should be on the preceding line. Why was the order of the debug statement and the optimizationTracker printout changed?